docs/registering-a-control-handler-function.md

```diff
@@ -90,3 +93,133 @@ int main(void)
+
+More information on setting up a window and its messaging loop can be found at [Creating a Window](/windows/win32/learnwin32/creating-a-window).
+
+```C
+// CtrlHandler.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <Windows.h>
+#include <stdio.h>
+
+BOOL WINAPI CtrlHandler(DWORD fdwCtrlType)
+{
+    switch (fdwCtrlType)
+    {
+        // Handle the CTRL-C signal.
+    case CTRL_C_EVENT:
+        printf("Ctrl-C event\n\n");
+        Beep(750, 300);
+        return TRUE;
+
+        // CTRL-CLOSE: confirm that the user wants to exit.
+    case CTRL_CLOSE_EVENT:
+        Beep(600, 200);
+        printf("Ctrl-Close event\n\n");
+        return TRUE;
+
+        // Pass other signals to the next handler.
+    case CTRL_BREAK_EVENT:
+        Beep(900, 200);
+        printf("Ctrl-Break event\n\n");
+        return FALSE;
+
+    case CTRL_LOGOFF_EVENT:
+        Beep(1000, 200);
+        printf("Ctrl-Logoff event\n\n");
+        return FALSE;
+
+    case CTRL_SHUTDOWN_EVENT:
+        Beep(750, 500);
+        printf("Ctrl-Shutdown event\n\n");
+        return FALSE;
+
+    default:
+        return FALSE;
+    }
+}
+
+LRESULT CALLBACK WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    switch (uMsg)
+    {
+    case WM_QUERYENDSESSION:
+    {
+        // Check `lParam` for which system shutdown function and handle events.
+        // See https://docs.microsoft.com/windows/win32/shutdown/wm-queryendsession
+        return TRUE; // Respect user's intent and allow shutdown.
+    }
+    case WM_ENDSESSION:
+    {
+        // Check `lParam` for which system shutdown function and handle events.
+        // See https://docs.microsoft.com/windows/win32/shutdown/wm-endsession
+        return 0; // We have handled this message.
+    }
+    default:
+        return DefWindowProc(hwnd, uMsg, wParam, lParam);
+    }
+}
+
+int main(void)
+{
+    WNDCLASS sampleClass{ 0 };
+    sampleClass.lpszClassName = TEXT("CtrlHandlerSampleClass");
+    sampleClass.lpfnWndProc = WindowProc;
+
+    if (!RegisterClass(&sampleClass))
+    {
+        printf("\nERROR: Could not register window class");
+        return 2;
+    }
+
+    HWND hwnd = CreateWindowEx(
+        0,
+        sampleClass.lpszClassName,
+        TEXT("Console Control Handler Sample"),
+        0,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        NULL,
+        NULL,
+        NULL,
+        NULL
```

Comment:
introducing new sample code in 2021 that uses `NULL` 😦

Reply:
I only did it because the block was labeled as C. I can adjust that...